components/common/src/templating/hooks.rs

```diff
@@ -440,54 +440,81 @@ pub struct HookOutput<'a> {
 
 impl<'a> HookOutput<'a> {
     fn new(stdout_log: &'a Path, stderr_log: &'a Path) -> Self {
-        HookOutput { stdout_log_file: stdout_log,
-                     stderr_log_file: stderr_log, }
+        Self { stdout_log_file: stdout_log,
+               stderr_log_file: stderr_log, }
+    }
+
+    pub fn stdout(&self) -> Result<BufReader<File>> {
+        Ok(BufReader::new(File::open(&self.stdout_log_file)?))
     }
 
-    pub fn stdout(&self) -> Option<BufReader<File>> {
-        match File::open(&self.stdout_log_file) {
-            Ok(f) => Some(BufReader::new(f)),
-            Err(_) => None,
+    pub fn stdout_str(&self) -> Option<String> {
```

Comment:
The reader version still [exists](https://github.com/habitat-sh/habitat/blob/b2df8cc0508226dd26cb9bf5621494074991a894/components/common/src/templating/hooks.rs#L447). We currently only read in all hook output for healthchecks. I like the idea of a tunable config parameter, but it is not obvious to me which part of the output we should keep? Maybe just a flag to turn on or off capturing healthcheck output? Or we could even go so far as having a prefix that indicates if a line should be kept? Ultimately though, I think my suggestion would be to hold off on any of these changes until we have a need.

The reason I choose an `Option` was similar to the motivation in this [comment](https://github.com/habitat-sh/habitat/pull/6905#discussion_r319513306). It allowed me to centrailize the error handeling and avoid duplicate code. If this is a pattern we generally want to avoid happy to change it.

Reply:
The `None` variant signifies there was an error preventing us from getting the output. We do pass that information on to automate. The stdout passed [here](https://github.com/habitat-sh/habitat/blob/b2df8cc0508226dd26cb9bf5621494074991a894/components/sup/src/event.rs#L235) is an `Option` as well, and will be `null` in the protobuf message if there was an error. If there was simply no output we would send an empty string instead of `null`. Now if we wanted to tell automate what the error was I agree this should be a `Result`.

Reply:
After talking with @danielsdeleo, we decided it will be important to limit healthcheck output and will limit it to 1k by default.